File: src/main/java/trap/report/CommandLineAppStartupRunner.java
package trap.report;

import lombok.RequiredArgsConstructor;
import org.jspecify.annotations.NonNull;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

@Component
@Profile("!test")
@RequiredArgsConstructor
public class CommandLineAppStartupRunner implements CommandLineRunner {
    private final ReportService reportService;

    @Override
    public void run(String @NonNull ... args) throws Exception {
        reportService.generateExcelFile();
    }
}

File: src/main/java/trap/report/CommandLineAppStartupRunner.kt
package trap.report

import org.springframework.boot.CommandLineRunner
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component

@Profile("!test")
@Component
class CommandLineAppStartupRunner : CommandLineRunner {
    private val reportHelper: ReportHelper = ReportHelper()

    @Throws(Exception::class)
    override fun run(vararg args: String) {
        reportHelper.doItAll()
    }
}
